CI: Fail when check_compliance finds errors

nashif opened the pull request.

nashif committed:
sanitycheck: fail on check_compliance issues

We were forcing sucess even if check_compliance had failed.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

carlescufi commented:
CC @myrdyr

dbkinder reviewed (commented):
I suspect you want me to ignore this PR :)  blah.

nashif committed:
sanitycheck: remove special ansi charachters from xml

XML does not like special ansi charachters (coming from logger), so
remove them.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
